packages/opentelemetry-core/src/common/time.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { performance } from 'perf_hooks';
+
+const NANOSECOND_DIGITS = 9;
+const SECOND_TO_NANOSECONDS = Math.pow(10, NANOSECOND_DIGITS);
+
+// Converts a number to HrTime
+function numberToHrtime(time: number): types.HrTime {
+  // Decimals only.
+  const seconds = Math.trunc(time);
+  // Round sub-nanosecond accuracy to nanosecond.
+  const nanos =
+    Number((time - seconds).toFixed(NANOSECOND_DIGITS)) * SECOND_TO_NANOSECONDS;
+  return [seconds, nanos];
+}
+
+// Returns an hrtime calculated via performance component.
+export function hrTime(performanceNow?: number): types.HrTime {
+  const timeOrigin = numberToHrtime(performance.timeOrigin);
```

Comment:
Just re-iterating comment above that `performance.timeOrigin` is in milliseconds (see https://www.w3.org/TR/hr-time-2/#dom-performance-timeorigin and https://www.w3.org/TR/hr-time-2/#dom-domhighrestimestamp). So the function above needs to be updated to reflect that (basically just divide by 1000 first).